util/src/main/java/tech/pegasys/teku/util/async/DelayedExecutorAsyncRunner.java

```diff
@@ -58,15 +56,23 @@ public static DelayedExecutorAsyncRunner create(Executor executor) {
   <U> SafeFuture<U> runAsync(final Supplier<SafeFuture<U>> action, final Executor executor) {
     final SafeFuture<U> result = new SafeFuture<>();
     try {
-      executor.execute(() -> propagateResult(action.get(), result));
-    } catch (RejectedExecutionException ex) {
+      executor.execute(() -> runTask(action, result));
+    } catch (final RejectedExecutionException ex) {
       LOG.debug("shutting down ", ex);
     } catch (final Throwable t) {
       result.completeExceptionally(t);
     }
     return result;
   }
 
+  private <U> void runTask(final Supplier<SafeFuture<U>> action, final SafeFuture<U> result) {
+    try {
+      action.get().propagateTo(result);
+    } catch (final Throwable t) {
```

Comment:
Looks good to me, but isn't this an anti-pattern? and is there any better way of doing this potentially?

Reply:
It's not anti-pattern though you do want to be careful when catching literally everything - you generally don't want to just log and suppress `Error` type exceptions but in this case we're just redirecting them back to the caller instead of them going straight to the uncaught exception handler.

Normally we just hide this pattern behind `SafeFuture.fromRunnable` and it turns out we also have `SafeFuture.ofComposed` which does exactly what we need, so I've switched to that.